hack/bootstrap.sh

```diff
@@ -24,8 +24,8 @@ if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
 fi
 
 sandbox_root=${SANDBOX_DIR}/default/root
-sandbox_hash_x86_64="a37eefec23911f2bb089f4eace5acfa83ba5f852"
-sandbox_hash_aarch64="b13b1bea08eef143be6a1e059fff43c765f466ed"
+sandbox_hash_x86_64="e52ec652d237d4fb1d89f23fa37453c2622e0502"
+sandbox_hash_aarch64="e52ec652d237d4fb1d89f23fa37453c2622e0502"
```

Comment:
From this commit forward, the hashes are always identical. I don't understand why that is not the case from the moment you start looking at `rpm/BUILD.bazel` only?

Either way, two considerations:

* since you're hashing a file that contains definitions for both architectures, you shouldn't need to keep track of separate hashes;
* the directory where the sandbox is created should probably be arch-qualified, to avoid the situation where an x86_64 sandbox is mistaken for an aarch64 one or vice versa;
* the two calls to `rm ${SANDBOX_DIR} -rf` in `hack/rpm-deps.sh` seem unnecessary - I think you should be able to drop both of them, but at the very least a single one should be enough.

Reply:
> From this commit forward, the hashes are always identical. I don't understand why that is not the case from the moment you start looking at `rpm/BUILD.bazel` only?

Yes this is a left-over from my initial experiments where I really created a shasum of the chroot enviroments. I will collapse it back to one shasum.




> * since you're hashing a file that contains definitions for both architectures, you shouldn't need to keep track of separate hashes;

Indeed. :+1: 

> * the directory where the sandbox is created should probably be arch-qualified, to avoid the situation where an x86_64 sandbox is mistaken for an aarch64 one or vice versa;

I am not sure how that can happen since the build root is based on the node architecture.

> * the two calls to `rm ${SANDBOX_DIR} -rf` in `hack/rpm-deps.sh` seem unnecessary - I think you should be able to drop both of them, but at the very least a single one should be enough.

Will check that. Thanks.

Reply:
> On a purely personal note, I find it easier to review changes when there is no "back and forth", with things going in one direction with one commit only to swing back in the opposite one a few commits later: even when the second change is explained in the commit message, it seems preferable to not include the "false starts" in the history at all. But again, that's just how I'm used to doing things and not necessarily the One Correct Way™ ;)
> 
> > >

Definitely. Just kept the history while being in "draft" mode. Should look better now.

Reply:
> Maybe it can't, but in the script there is a check for the existence of the directory and since the name is based solely on the hash itself it feels a bit less fragile to also have the architecture name in there... It's entirely possible that I'm worrying way too much about this though ;)

Did not address that. I think we are fine ™ :)

Reply:
> Definitely. Just kept the history while being in "draft" mode. Should look better now.

It does! Thank you :)

> Did not address that. I think we are fine ™ :)

Let's cross our fingers and hope you are right then ;)